src/test/java/tests/UpdateConfigTest.java

```diff
@@ -0,0 +1,118 @@
+package tests;
+
+import org.junit.Test;
+import prefs.UpdateConfig;
+import util.Version;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+public class UpdateConfigTest {
+    @Test
+    public void updateConfigSetLastUpdateDownloadStatus_getValueByReflection_correctValue() {
+        UpdateConfig updateConfig = new UpdateConfig();
+
+        try {
+            Class<?> updateConfigClass = updateConfig.getClass();
+
+            Field lastUpdateDownloadStatusField = updateConfigClass.getDeclaredField("lastUpdateDownloadStatus");
+            lastUpdateDownloadStatusField.setAccessible(true);
+
+            updateConfig.setLastUpdateDownloadStatus(true);
+            assertTrue((boolean) lastUpdateDownloadStatusField.get(updateConfig));
+
+            updateConfig.setLastUpdateDownloadStatus(false);
+            assertFalse((boolean) lastUpdateDownloadStatusField.get(updateConfig));
+        } catch (NoSuchFieldException e) {
```

Comment:
These exceptions can go into the method signature I think. Makes the test cleaner.